ocaml/typing/typedtree.mli

```diff
@@ -110,16 +110,18 @@ and 'k pattern_desc =
             See {!Types.row_desc} for an explanation of the last parameter.
          *)
   | Tpat_record :
-      (Longident.t loc * Types.label_description * value general_pattern) list *
-        closed_flag ->
-      value pattern_desc
+      (Longident.t loc * Types.label_description * value general_pattern
+        * Types.alloc_mode option) list * closed_flag -> value pattern_desc
         (** { l1=P1; ...; ln=Pn }     (flag = Closed)
             { l1=P1; ...; ln=Pn; _}   (flag = Open)
 
             Invariant: n > 0
+
+            [alloc_mode option] only not none when it is a float record
```

Comment:
When you extract a `float` out of a float record, you need to allocate the space for the float. The `alloc_mode` is the mode the new float will be allocated at, and it will be either `global` or `local` depending on the newly-allocated `float` will be used.

Reply:
Yes, if you have a float-record, all `alloc_mode` will be `Some`. The new representation you suggested reflects that, but on the hand doesn't guarantee the list of `alloc_mode` and the list of `element` match in length.

Reply:
One minor reason to prefer the current representation: `Record_float` is read out of each label/field's `lbl_repres`, giving the superficial impression that this is a per-field property.

Reply:
Good point about keeping the lengths in sync. I don't think it matters much which representation is chosen, as long as these details appear in the comments, both about what the alloc_mode means and also about any expected invariants.